### PR TITLE
New feature: it fetches the instrument data \`unitsReported\` and fin…

### PR DESCRIPTION
**Pull request by Khosiyat**
## Description
This pull request introduces a new feature related to fetching `unitsReported` for the Y-axis.

## Changes Made
-  Updated code to address feedback in pull request #79. Please, see two comments provided in the code block.

[Link of the JIRA Task](https://biologging-data-lu.atlassian.net/browse/BIOSENSOR-79?atlOrigin=eyJpIjoiNWNhYzQ0ZTRiNTgwNDI5MjhlYzE0NjZiZTRiNzYwYjciLCJwIjoiaiJ9)

### CODE REVIEW

**Comment by Khosiyat** [comment]
[Link of the JIRA Task](https://biologging-data-lu.atlassian.net/browse/BIOSENSOR-79?atlOrigin=eyJpIjoiNWNhYzQ0ZTRiNTgwNDI5MjhlYzE0NjZiZTRiNzYwYjciLCJwIjoiaiJ9)

File: biologging-sensor-data/src/api/instrumentApi.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// ./src/api/instrumentApi.ts
+import { get } from './api'; // Assuming you have the API functions defined as you provided
+
+export const fetchInstrumentData = async (instrumentID) => {
```
[the remark this answers is not in the document]
Comment (Khosiyat) [reply biologging-sensor-data/src/api/instrumentApi.ts:4]: Yeah, this part is the result of my first experience using AI autocomplete. You're right it is unnecessary.

File: biologging-sensor-data/src/components/graphs/line/LineGraph.tsx [hunk 2]
```diff
@@ -36,35 +37,37 @@ interface LineData {
...
-      },
-    },
-  });
+  const [unit, setUnit] = useState<string | null>(null);
+  const instrumentID = '1c30-47f5-9f1d-a3e7dc1a1'; // Now the Y-Axis satisfies our requirement. However, we should replace this instrument ID with dynamic instrumentID.
+
+  useEffect(() => {
+    const fetchInstrumentData = async () => {
+      try {
+        const response = await axios.get<InstrumentData>(`http://canmove-dev.ekol.lu.se:8080/biologgingAPI/v1/instrument/${instrumentID}`); // Maybe seperate API componenet is created for this instrument API.
```
[the remark this answers is not in the document]
Comment (Khosiyat) [reply biologging-sensor-data/src/components/graphs/line/LineGraph.tsx:56]: I was thinking the same, so I created `instrumentApi.ts` and experimented. But, I could not achieve the desired result. Will try again.

File: biologging-sensor-data/src/components/graphs/line/LineGraph.tsx [hunk 3]
```diff
@@ -36,35 +37,37 @@ interface LineData {
...
-      y: {
-        title: {
-          display: true,
-          text: sensor,
-        },
-      },
-    },
-  });
+  const [unit, setUnit] = useState<string | null>(null);
+  const instrumentID = '1c30-47f5-9f1d-a3e7dc1a1'; // Now the Y-Axis satisfies our requirement. However, we should replace this instrument ID with dynamic instrumentID.
```
[the remark this answers is not in the document]
Comment (zrohacova) [reply biologging-sensor-data/src/components/graphs/line/LineGraph.tsx:51]: @Khosiyat I think I wrote to you on Slack, you take the instrumentId from the event object that goes as a prop to the class

File: biologging-sensor-data/src/components/graphs/line/LineGraph.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (zrohacova) [reply biologging-sensor-data/src/components/graphs/line/LineGraph.tsx:56]: Yeah, just check the api/datasets/api.ts (you can also rename instrumentApi.ts to api.ts) and you can see how it is done there, for example method getDataset

**Pull request closed without merging**